### PR TITLE
Collector gem counter

### PR DESCRIPTION
**Pull request by Hamms**
[Spec](https://docs.google.com/document/d/1vqi9eGbytLNM3_niL1sfBdoo73Yl8gMdSf0ONdkl8UM/edit)

Add a Gem Counter component to the visualization button area for Collector levels:

image

Component displays the minimum required number of gems to collect to complete the level (which is often but not always going to be the maximum number of gems) and updates realtime as code is run to show a running count (plus a little green checkmark when you hit the required min):

gemcounter

### CODE REVIEW

**Comment by Hamms** [comment]
fyi @ryansloan

File: apps/src/maze/CollectorGemCounter.jsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import { connect } from 'react-redux';
+import color from '../util/color';
+import msg from './locale';
+
+const styles = {
+  container: {
+    display: 'inline-block',
+    fontFamily: '"Gotham 5r"',
+    paddingRight: 10,
+    pointerEvents: "none",
```
Comment (joshlory) [inline apps/src/maze/CollectorGemCounter.jsx]: Tiny nit: mixing quote types here (and a few places below).

File: apps/src/maze/CollectorGemCounter.jsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    color: 'lightgreen',
+    fontSize: 18,
+    left: 5,
+    lineHeight: '25px',
+    position: 'relative',
+    top: 7,
+  },
+};
+
+export class CollectorGemCounter extends React.Component {
```
Comment (joshlory) [inline apps/src/maze/CollectorGemCounter.jsx:42]: React Storybook entry?

File: apps/src/maze/CollectorGemCounter.jsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    minRequired: React.PropTypes.number,
+  }
+
+  static defaultProps = {
+    minRequired: 1,
+  }
+
+  render() {
+    const showCheckmark = this.props.currentCollected >= this.props.minRequired;
+    const gemUrl = this.props.assetUrl('media/skins/collector/gem.png');
```
Comment (joshlory) [inline apps/src/maze/CollectorGemCounter.jsx]: You should be able to use `import assetUrl from '@cdo/apps/code-studio/assetUrl';` now!

https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/code-studio/assetUrl.js

File: apps/src/maze/CollectorGemCounter.jsx (hunk 2 above)
Comment (Hamms) [reply apps/src/maze/CollectorGemCounter.jsx:42]: I originally had one, but had to remove it; Storybook doesn't play nicely with app-specific i18n ('maze/locale') unless we do [special override work](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/webpack.js#L24-L28) and that seemed like too much of a refactor to justify

Comment (joshlory) [reply apps/src/maze/CollectorGemCounter.jsx:42]: Ah, yuck 🙁.  Another option would be to put "goal" in `common.js`.

File: apps/src/maze/CollectorGemCounter.jsx (hunk 3 above)
Comment (Hamms) [reply apps/src/maze/CollectorGemCounter.jsx]: that is _so much_ better! Thanks!

File: apps/src/maze/CollectorGemCounter.jsx (hunk 2 above)
Comment (Hamms) [reply apps/src/maze/CollectorGemCounter.jsx:42]: yeah, but then I'd be polluting `common.js` with a collector-specific phrase _just_ to add a storybook entry.

I did a bit of looking around, and it turns out _none_ of our React components with app-specific translations have Storybook entries, probably for this same reason. I think the solution to that is probably to add a tech debt item to look into why storybook is being a grouch about this.

**Pull request merged**